### PR TITLE
Decouple the two /trigger controllers

### PR DESCRIPTION
**Pull request by hennevogel**
We inherit from TriggerController in TriggerWorkflowController just for one `before_action`. We don't use anything else from TriggerController. This does not make much sense anymore.

### CODE REVIEW

**Comment by danidoni** [comment]
https://trello.com/c/s3rXELVj/316-stop-using-inheritance-on-workflowtriggercontroller

File: src/api/app/controllers/trigger_workflow_controller.rb [hunk 1]
```diff
@@ -35,6 +44,15 @@ def create
...
   private
 
+  def set_token
+    @token = ::TriggerControllerService::TokenExtractor.new(request).call
+    raise InvalidToken, 'No valid token found' unless @token
+  end
+
+  def pundit_user
+    @token.executor
+  end
```
Comment (danidoni) [inline src/api/app/controllers/trigger_workflow_controller.rb:54]: We can extract those two methods into a module and share it with `TriggerController` also

Comment (danidoni) [reply src/api/app/controllers/trigger_workflow_controller.rb:54]: To achieve DRYness. The next time someone needs to change those methods they will start to diverge. Having them in a single place will make people think about all the use cases.

Comment (hennevogel) [reply src/api/app/controllers/trigger_workflow_controller.rb:54]: If that "next time" comes we can optimize it. Or in other words: f**k the future :)

Comment (danidoni) [reply src/api/app/controllers/trigger_workflow_controller.rb:54]: All right, then, f**ck the future.

**Review by danidoni — APPROVED** [review]

**Pull request merged**